File: demo/service/reservation_service.c
#include "reservation_service.h"

int reserverProduit(Produit* produit, int duree, Reservation* res) {
    if (produit->etat == INDISPONIBLE) {
        return 0;
    }

    produit->etat = INDISPONIBLE;
    res->produit = produit;
    res->duree = duree;
    return 1;
}

File: demo/service/ReservationService.js
const ProduitIndisponibleException = require("../exception/ProduitIndisponibleException");
const Reservation = require("../reservation/Reservation");

class ReservationService {
  reserver(produit, duree) {
    if (!produit.disponible) {
      throw new ProduitIndisponibleException("Ce produit n'est plus disponible.");
    }

    produit.disponible = false;
    return new Reservation(produit, duree);
  }
}

module.exports = ReservationService;
